File: 30-day-leetcode/11_Container_with_most_water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int lp=0;
        int rp=height.size()-1;
        int maxVol=INT_MIN;

        while(lp<rp){
            int width=rp-lp;
            int ht=min(height[rp],height[lp]);
            int currVol=width*ht;
            maxVol=max(maxVol,currVol);
            if(height[rp]>height[lp]){
                lp++;
            }
            else{
                rp--;
            }
        }
        return maxVol;
    }
};

File: TwoPointers/11ContainerWithMostWater.ts
function maxArea(height: number[]): number {
    let l=0;
    let r=height.length-1;
    let maxarea=0;
    while(l<r){
        maxarea = Math.max(Math.min(height[l],height[r])*(r-l),maxarea); 
        if(height[l]>height[r]){
            r--;
            
        }
        else{
            l++;
        }
    }
    return maxarea
};